modules/reboot/src/main/java/org/mule/runtime/module/reboot/internal/AbstractMuleContainerWrapper.java

```diff
@@ -82,8 +82,20 @@ private boolean waitAllConfigurersReady() {
   }
 
   public void dispose() {
+    RuntimeException disposalException = null;
     for (BootstrapConfigurer bootstrapConfigurer : bootstrapConfigurers) {
-      bootstrapConfigurer.dispose();
+      try {
+        bootstrapConfigurer.dispose();
+      } catch (Throwable t) {
+        // Records the first exception but don't throw just yet, to let the other Configurers to run their disposal.
+        if (disposalException == null) {
+          disposalException = new RuntimeException(t);
+        }
+      }
+    }
+
+    if (disposalException != null) {
+      throw disposalException;
```

Comment:
would be better to throw a new exception, with all its causes as suppressed, that way if there are many failures, all will be logged instead of just the first one.